tests/api/crud/test_alerts.py

```diff
@@ -104,33 +94,198 @@ async def test_validate_alert_name(
+            ),
+            True,
+        ),
+        (
+            "trigger",
+            alert_objects.AlertTrigger(
+                events=[alert_objects.EventKind.DATA_DRIFT_DETECTED]
+            ),
+            True,
+        ),
+        (
+            "criteria",
+            alert_objects.AlertCriteria(
+                count=5,
+                period="10m",
+            ),
+            True,
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "force_reset",
+    [False, True],
+)
+async def test_alert_reset_with_fields_updates(
+    db: sqlalchemy.orm.Session,
+    modify_field,
+    modified_value,
+    should_reset,
+    force_reset,
+    k8s_secrets_mock: tests.api.conftest.K8sSecretsMock,
+):
+    project = "project-name"
+    alert_name = "failed-alert"
+    alert_summary = "The job has failed"
+    alert_reset_policy = alert_objects.ResetPolicy.MANUAL
+    alert_entity = alert_objects.EventEntities(
+        kind=alert_objects.EventEntityKind.JOB,
+        project=project,
+        ids=[123],
+    )
+    event_kind = alert_objects.EventKind.FAILED
+    alert_trigger = alert_objects.AlertTrigger(events=[event_kind])
+
+    alert_data = _generate_alert_data(
+        project=project,
+        name=alert_name,
+        entity=alert_entity,
+        summary=alert_summary,
+        trigger=alert_trigger,
+        reset_policy=alert_reset_policy,
+    )
+
+    # store the initial alert
+    server.api.crud.Alerts().store_alert(
+        session=db,
+        project=project,
+        name=alert_name,
+        alert_data=alert_data,
+    )
+
+    # activate the alert
+    event = alert_objects.Event(kind=event_kind, entity=alert_entity)
+    await fastapi.concurrency.run_in_threadpool(
+        server.api.crud.Alerts().process_event_no_cache,
+        db,
+        event.kind,
+        event,
+    )
+    alert = server.api.crud.Alerts().get_enriched_alert(
+        session=db,
+        project=project,
+        name=alert_name,
+    )
+    assert alert.state == alert_objects.AlertActiveState.ACTIVE
+
+    # modify the alert data based on the parameterized field
+    setattr(alert_data, modify_field, modified_value)
+
+    # store the modified alert
+    server.api.crud.Alerts().store_alert(
+        session=db,
+        project=project,
+        name=alert_name,
+        alert_data=alert_data,
+        force_reset=force_reset,
+    )
+
+    # fetch the updated alert
+    alert = server.api.crud.Alerts().get_enriched_alert(
+        session=db,
+        project=project,
+        name=alert_name,
+    )
+
+    # validate the state based on whether it should have reset
+    expected_state = (
+        alert_objects.AlertActiveState.INACTIVE
+        if should_reset or force_reset
+        else alert_objects.AlertActiveState.ACTIVE
+    )
+    assert alert.state == expected_state
+
+    # reset the alert so the next test starts clean
+    server.api.crud.Alerts().reset_alert(session=db, project=project, name=alert_name)
```

Comment:
If one of the `assert`s above fails we would never get here.
You can maybe use a fixture to make sure it is always called.